app/models/orchestration_stack.rb

```diff
@@ -31,10 +31,7 @@ def tenant_identity
   end
 
   def directs_and_indirects(direct_attrs)
-    return send(direct_attrs) if descendants.blank?
-
-    indirects = descendants.inject([]) { |arr, child| arr << child.send(direct_attrs) }
-    (indirects << send(direct_attrs)).flatten
+    MiqPreloader.preload_and_map(subtree, direct_attrs)
```

Comment:
@matthewd I didn't know you could do a `:through => scope`

I must be doing this wrong, I'm getting the following error:
```
Failure/Error: expect(root_stack.vms.size).to eq(2)
     
ActiveRecord::HasManyThroughAssociationNotFoundError:
Could not find the association :subtree in model ManageIQ::Providers::CloudManager::OrchestrationStack
```